### ISSUE TITLE
[Bug]: BufferLineGroupClose ungrouped is removing the pin of pinned tabs

### ISSUE THREAD

**Issue by otavioschwanck** [akinsho/bufferline.nvim#695]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

[Bug]: BufferLineGroupClose ungrouped is removing the pin of pinned tabs

### What did you expect to happen?

The pins should be keeped

### Config

```
return {
  {
    "akinsho/bufferline.nvim", branch = 'dev',
    lazy = false,
    keys = {
      { "<leader>bp", "<Cmd>BufferLineTogglePin<CR>",            desc = "Toggle pin" },
      { "<leader>bk", "<Cmd>BufferLineGroupClose ungrouped<CR>", desc = "Delete non-pinned buffers" },
      { "<leader>bK", "<Cmd>bufdo bwipeout<CR>",                 desc = "Kill all buffers" },
      { "<leader>bn", ":e ~/.nvim-scratch<CR>",                  desc = "Open Scratch Buffer" },
      { "<C-s>",      "<cmd>BufferLineTogglePin<cr>",            desc = "Prev Buffer" },
      { "[b",         "<cmd>BufferLineCyclePrev<cr>",            desc = "Prev Buffer" },
      { "]b",         "<cmd>BufferLineCycleNext<cr>",            desc = "Next Buffer" },
      { "H",          "<cmd>BufferLineCyclePrev<cr>",            desc = "Prev Buffer" },
      { "L",          "<cmd>BufferLineCycleNext<cr>",            desc = "Next Buffer" },
      { "[B",         "<cmd>BufferLineMovePrev<cr>",             desc = "Switch Prev Buffer" },
      { "]B",         "<cmd>BufferLineMoveNext<cr>",             desc = "Switch Next Buffer" },
      { "<C-h>",      "<cmd>BufferLineMovePrev<cr>",             desc = "Switch Prev Buffer" },
      { "<C-l>",      "<cmd>BufferLineMoveNext<cr>",             desc = "Switch Next Buffer" },
    },
    opts = {
      options = {
        diagnostics = "nvim_lsp",
        always_show_bufferline = false,
        persist_buffer_sort = true,
        offsets = {
          {
            filetype = "neo-tree",
            text = "Neo-tree",
            highlight = "Directory",
            text_align = "left",
          },
        },
      },
    },
  }
}
```

### commit

894d0e80583cacd27facdb6676a6b08764662537

**Comment by akinsho** [comment]
Seems like https://github.com/akinsho/bufferline.nvim/commit/894d0e80583cacd27facdb6676a6b08764662537#diff-47fabfe452ff7a153117c6d9b99f2ef62455cb8ae325dd77fd62d85fe3d10d4dR138 line in the PR broke things. It shouldn't be resetting the pinned values *every time a group is closed* only if the *pinned* group is closed

**Comment by otavioschwanck** [comment]
> Seems like [894d0e8#diff-47fabfe452ff7a153117c6d9b99f2ef62455cb8ae325dd77fd62d85fe3d10d4dR138](https://github.com/akinsho/bufferline.nvim/commit/894d0e80583cacd27facdb6676a6b08764662537#diff-47fabfe452ff7a153117c6d9b99f2ef62455cb8ae325dd77fd62d85fe3d10d4dR138) line in the PR broke things. It shouldn't be resetting the pinned values _every time a group is closed_ only if the _pinned_ group is closed

The persist should only persist, it should not have effect on the active pins, right?

**Comment by akinsho** [comment]
fixed by #696
